juju_verify/verifiers/ovn_central.py

```diff
@@ -0,0 +1,399 @@
+#
+# juju-verify is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# juju-verify is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see https://www.gnu.org/licenses/.
+"""ovn-central verification."""
+import logging
+from collections import defaultdict
+from typing import Any, Dict, List, NamedTuple, Optional
+
+import yaml
+from juju.unit import Unit
+
+from juju_verify.exceptions import JujuVerifyError
+from juju_verify.utils.action import data_from_action
+from juju_verify.utils.unit import run_action_on_units
+from juju_verify.verifiers.base import BaseVerifier
+from juju_verify.verifiers.result import Result, Severity, checks_executor
+
+logger = logging.getLogger(__name__)
+
+StatusCollectionType = Dict[str, "UnitClusterStatus"]
+
+
+class ClusterStatus:  # pylint: disable=R0902
+    """Object representation of OVN cluster status.
+
+    Attributes of this object represent status of a single (Southbound or Northbound)
+    cluster as retrieved from 'cluster-status' action of ovn-central charm unit. Example
+    of an input data for initialization of this class:
+
+    .. code-block:: yaml
+
+        Cluster ID: 567e7225-369e-40d6-abf8-9b442bbcd18b
+        Server ID: 16335def-c21e-404c-b123-8337b3013c07
+        Address: ssl:10.5.2.232:6644
+        Status: cluster member
+        Role: follower
+        Term: 34
+        Leader: dbdb
+        Vote: dbdb
+        Log: '[66, 66]'
+        Entries not yet committed: 0
+        Entries not yet applied: 0
+        Servers:
+          dbdb:
+            Address: ssl:10.5.0.144:6644
+            Unit: ovn-central/7
+          f1a2:
+            Address: ssl:10.5.3.200:6644
+            Unit: ovn-central/10
+          '1633':
+            Address: ssl:10.5.2.232:6644
+            Unit: ovn-central/6
+    """
+
+    def __init__(self, raw_status: str) -> None:
+        """Initialize Cluster status class.
+
+        For more info about the purpose and input data format see class documentation.
+
+        :param raw_status: YAML string containing status of a single OVN cluster.
+        """
+        try:
+            status_dict = yaml.safe_load(raw_status)
+            if not isinstance(status_dict, dict):
+                raise ValueError("Input data is not a YAML dictionary.")
+        except (yaml.YAMLError, ValueError) as exc:
+            raise JujuVerifyError(
+                f"Failed to load OVN status as YAML:\n{raw_status}"
+            ) from exc
+
+        try:
+            self.cluster_id = status_dict["Cluster ID"]
```

Comment:
I think that using `NamedTuple` in this case will make much more sense.

- comparing will be easier
- if for example "Server ID" is not a string, but integer it will failed during loading this status, rather than during `check_leader_consistency`